src/H5private.h

```diff
@@ -366,7 +375,7 @@
  * the entire library.
  */
 #define H5_IS_KNOWN_BUFFER_OVERFLOW(skip, ptr, size, buffer_end)                                             \
-    (skip ? false : ((ptr) + (size)-1) > (buffer_end))
+    (skip ? false : H5_IS_BUFFER_OVERFLOW(ptr, size, buffer_end))
```

Comment:
Update this macro to use `H5_IS_BUFFER_OVERFLOW` when not skipping buffer overflow check